man/partition.d.xml

```diff
@@ -0,0 +1,372 @@
+              </row>
+
+              <row>
+                <entry><constant>root-verity</constant></entry>
+                <entry>Verity data for the root file system partition for the local architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-secondary</constant></entry>
+                <entry>Root file system partition of the secondary architecture of the local architecture; usually the matching 32bit architecture for the local 64bit architecture)</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-secondary-verity</constant></entry>
+                <entry>Verity data for the root file system partition of the secondary architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86</constant></entry>
+                <entry>Root file system partition for the x86 (32bit, aka i386) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86-verity</constant></entry>
+                <entry>Verity data for the x86 (32bit) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86-64</constant></entry>
+                <entry>Root file system partition for the x86_64 (64bit, aka amd64) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-x86-64-verity</constant></entry>
+                <entry>Verity data for the x86_64 (64bit) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm</constant></entry>
+                <entry>Root file system partition for the ARM (32bit) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm-verity</constant></entry>
+                <entry>Verity data for the ARM (32bit) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm64</constant></entry>
+                <entry>Root file system partition for the ARM (64bit, aka aarch64) architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-arm64-verity</constant></entry>
+                <entry>Verity data for the ARM (64bit, aka aarch64) root file system partition</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-ia64</constant></entry>
+                <entry>Root file system partition for the ia64 architecture</entry>
+              </row>
+
+              <row>
+                <entry><constant>root-ia64-verity</constant></entry>
+                <entry>Verity data for the ia64 root file system partition</entry>
+              </row>
+            </tbody>
+          </tgroup>
+        </table>
```

Comment:
could we have an entry for the "var" partition ? it is pretty common to have that subtree on a separate partition...

Reply:
so far we didn't define that because /var unlike the others listed is strictly only usable in the context of a specific installation. swap, /srv and /home can be somewhat reasonably shared between installations, but /var could not possibly be shared. Thus while we can get away with enumerating the other ones and just mounting them when we find them I don't think this would work for /var, because if you have two distros installed onto the same HDD then bad things would happen when they both mount the same /var.

That said, I think we can deal with that in a reasonable way now: only auto-mount the var partition if the GPT partition uuid of it matches a hash of the machine ID of the system. This way we can bind a specific /var to a specific /etc/machine-id and thus would never be confused when installing two distros onto the same HDD, because those distros will have a different /etc/machine-id. Now you might wonder: how would I prepare images for that? Wouldn't that mean that the machine ID would have to be set when the image is built rather than on first boot? systemd-repart is the way out of that actually: it allows you to either create the var partition on first boot automatically, and if that's not desirable already has a mechanism built in to initialize the partition UUID of any listed partition of its all zeros, already based on the hash of /etc/machine-id. Thus you could prepare your image with the machine ID unset and the var partition uuid set to zeroes, and on first boot we'd first assign a random machine ID and then automatically update the partition table with systemd-repart to the hashd uuid for it.

But anyway, that's for a different PR...

Reply:
@boucman see PR #14390

Reply:
thx